packages/core/lib/span.ts

```diff
@@ -151,4 +185,20 @@ export class SpanFactory {
       this.processor.add(spanEnded)
     }
   }
+
+  spanFromSpanInternal (span: SpanInternal): Span {
+    return {
```

Comment:
I don't think it's critical that `Span` is truly readonly because we don't ever reference its ID or trace ID — we always read those from the `SpanInternal` and that is truly readonly as it's never exposed to a user

i.e. if you mutate the `Span` that's fine because we will always ignore those mutations and I don't see why we'd ever change that behaviour since the `SpanInternal` is always our source of truth

There are a couple of ways to make data truly private without private class fields (we can't use those for compatibility reasons):

- [using symbols](https://www.typescriptlang.org/play?#code/JYOwLgpgTgZghgYwgAgMoAc4gJLmiOAG2QG8AoZZKCOAEwHsRCBPZYWgLmQGcwpQA5hSo0GTVn0QRsnHn0HDg3AGpF2yABQBKLgCN69QjRDCA9KeQQwCAHR2yAXzJlQkWFLSYced0lLDqOkYWNllefhAhSkCxEMkkGS5whUolVUJ1bT0DIyxHZwRGXlDUZgBbfUIuAFcQYABHapRucsrkAF40VsMNAHJ2Xq0yQpBi+OlaUorDGrrG5u7iTqnKvvGZQeGisDYVNUnF2YamnkWOrunCPrT9zeHCOG5uTyw2MvQjMohwZ4wsXDc8D85Eo6H4ADc4JAREFxMgANrsFaGAC6SXkkWEYOAkOhMWCrHh6wOlzRcgiUWQ2NxKHxcMRewyJMqZO0HQAfMh9IZjM5KCNwtUEGB6FBNNwvAD8EQuH9vG4CIQtP5KJQwAALJSI5mo84S-4+RU2djCNWa7hEqBSGTIwgovWSw1EGzE03IDVam5M232zr6+XSwjGxnsGy6UC0DT+qVQRVDShOYQCKyhTQ6cmCFWq6hgaqx93m7U+4SJyjJnbEtPoilZ6JWPMgAta4nFhN83bpTLp7m5Rsg7P1-Mei1epGLFHaEv5LajHZwc4gCAAdxeIA0JFCXF6cF6ABp3VaErJero9x39mnSCJc-n4IRuCgHMgHEMgA), though you can get around it by using `Object.getOwnPropertySymbols` so it's not _completely_ always private
- [using a weak map](https://www.typescriptlang.org/play?#code/JYOwLgpgTgZghgYwgAgMoAc4gJLmiOAG2QG8AoZZKCOAEwHsRCBPZYWgLmQGcwpQA5hSo0GTVn0QRsnHn0HDg3AGpF2yABQBKLgCN69QjRDCA9KeQQwCAHR2yAXzJlQkWFLSYced0lLDqOkYWNllefhAhSkCxEMkkGS5whUolVUJ1bT0DIyxHZxgAVxAEMGBGZARAyAwsAGFCOG5uTS1-SgRGXmRaODA4ZABeZBAIAHdkAHUaAGsAWTh0AB5a7zd4JAAaTyxcNwJCAD5tZ2irQqgQSsbmnavgAFt0IweIcBbVvegNlHJKDq6fEKpXoUE03C8X0uRC4nx8Bzaf3+lF6-Rs3CsGjAAAslNsIbt4UQtMJKE5ScgBFZQq0kvJIu1kSIwBcrqi4DYqWAsbjuFoAIQ2dgU8nIrnIeLSWi0uQRASM5HUFmXHp9DlcnlKAU2SUyEWnf5pNTSrLIfSGYwK-5K1mqtEanFawVGjK0E7Ipxk-JkTogbqrIaVaoQVYNJrcE4+wHIAbDUYTVYaEihLgAcjgqe2utkqd0mbYKmNrVIzNt8EIGOQDirJKAA)

Having said that, it's probably not worth changing this